File: Projects/DataProviderApp/app/src/main/java/com/example/dataproviderapp/dto/requests/CreateGpsDataRequest.kt
package com.example.dataproviderapp.dto.requests

data class CreateGpsDataRequest(
    val recDatetime: String,
    val tripId: ULong,
    val latitudeDeg: Double,
    val longitudeDeg: Double,
    val accuracyM: Float?,
    val speedKmh: Int?,
    val bearingDeg: Float
)

File: Projects/server/Contracts/Requests/CreateGpsDataRequest.cs
namespace server.Contracts.Requests
{
    public class CreateGpsDataRequest
    {
        public DateTime RecDateTime { get; set; }
        public ulong TripId { get; set; }
        public float LatitudeDeg { get; set; }
        public float LongitudeDeg { get; set; }
        public float? AccuracyM { get; set; }
        public uint? SpeedKmh { get; set; }
        public float? BearingDeg { get; set; }
    }
}
